### PR TITLE
Follows proper autoloading standards

### PR DESCRIPTION
**Pull request by bshaffer**
Addresses #3107:

 - Splits PHP classes in descriptor.php into separate files
 - Splits MapFieldIter and RepeatedFieldIter into separate files
 - Moves descriptor.php to Internal/functions.php
 - Moves all namespaced functions into Iternal/functions.php

### CODE REVIEW

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by TeBoring** [comment]
ok to test

**Comment by TeBoring** [comment]
FYI, jenkins test failed (I don't know why it doesn't show up here): https://grpc-testing.appspot.com/view/Protocol%20Buffers/job/protobuf_pull_request/1194/
The problem is that composer is pre-installed for jenkins, which hasn't been updated before the PR is submitted.
After fix the comment I left, you can merge the PR and send another PR to update the commit id:
https://github.com/google/protobuf/blob/master/jenkins/docker/Dockerfile#L177
https://github.com/google/protobuf/blob/master/jenkins/docker32/Dockerfile#L67

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by bshaffer** [comment]
@TeBoring done! Please take a look.

File: Makefile.am [hunk 1]
```diff
@@ -643,7 +651,7 @@ php_EXTRA_DIST=                                                       \
   php/src/Google/Protobuf/Internal/GPBUtil.php                        \
   php/src/Google/Protobuf/Internal/FieldOptions_CType.php             \
   php/src/Google/Protobuf/Internal/GPBDecodeException.php             \
-  php/src/Google/Protobuf/descriptor.php                              \
+  php/src/Google/Protobuf/Internal/functions.php                      \
```
Comment (TeBoring) [inline Makefile.am]: Still need this?

Comment (TeBoring) [inline Makefile.am]: Still need this?

**Comment by bshaffer** [comment]
Do you know why these tests failed?

**Comment by TeBoring** [comment]
I think it's because the composer.json needed by jenkins tests is updated. We need to submit this PR first and update the Dockerfile to use the latest composer.json.

**Comment by TeBoring** [comment]
I can send a fixing PR after this is submitted.

**Comment by michaelbausor** [comment]
@bshaffer @TeBoring I think this PR is causing an error because the `getFullClassName()` function referenced here https://github.com/google/protobuf/blob/master/php/src/Google/Protobuf/Internal/Descriptor.php#L132 is defined elsewhere. I am seeing this error:

>Fatal error: Call to undefined function Google\Protobuf\Internal\getFullClassName() in .../php_proto3/google-cloud-php/vendor/google/protobuf/php/src/Google/Protobuf/Internal/Descriptor.php on line 132

**Pull request merged**